### ISSUE TITLE
Can't import contacts VCF file

### ISSUE THREAD

**Issue by DarkSteve** [owncloud/contacts#385]
### Steps to reproduce
1. Click "Import" button
2. Choose VCF file
3. Receive "Invalid file type" error
### Expected behaviour

Contacts should be imported.
### Actual behaviour

"Invalid file type" error and nothing imported.
### Server configuration

FreeBSD 10.2

Apache 2.4

MySQL 5.6

PHP 5.6

ownCloud 9.0.1

Contacts 1.2 (released just minutes ago)

Fresh install of ownCloud
### Client configuration

Pale Moon 26.2.1 on Win10

Basically, I've been waiting to import my backed up Contacts VCF file since I reinstalled my server. Previously in Contacts 1.1, there was no error message or feedback, but nothing was imported. Now I'm getting an "Invalid file type" error, and still can't import the VCF file that was exported by ownCloud 8.2.

HELP!!

**Comment by DeepDiver1975** [comment]
Which browser?
What is the file name?

**Comment by DeepDiver1975** [comment]
> Pale Moon 26.2.1 on Win10

ah - mind giving a try with another browser?

**Comment by DarkSteve** [comment]
Yep. Actually, just after I wrote the Issue, I tested on IE11, Edge, and Seamonkey 2.40 (on Win10) and the built in browser and Fennec Fox on Cyanogenmod11.

I get the same error. This was a VCF I exported from ownCloud 8.2 before I reinstalled from scratch - have I pooched my ownCloud? Or do I have to do something with my VCF?

**Comment by DeepDiver1975** [comment]
Looks like the browser cannot properly detect the mimetype of the file.
Is the file named foobar.vcf? AFAIK the correct extension name is required

**Comment by DeepDiver1975** [comment]
> before I reinstalled from scratch -

Just because I'm curious - why did you reinstall from scratch?

**Comment by bobblkabb** [comment]
Same here with Firefox 45.
**Operating system:**
Ubuntu 14.04.4 LTS
**Web server:**
Apache 2.4.7
**Database:**
MySQL 5.5.47
**PHP version:**
5.5.9
**ownCloud version:**
9.0.1
**Contacts version:**
1.2.0.0

**Comment by DeepDiver1975** [comment]
@bobblkabb what is the name of the file you want to import?

**Comment by aspdye** [comment]
can we add some code, that adds **.vcf** if it is not in the filename? 😕

**Comment by pravussum** [comment]
Same here with FF38 and the file named "Standard.vcf" or "Contacts.vcf" respectively.

**Comment by bobblkabb** [comment]
bobblkabb.vcf

**Comment by DarkSteve** [comment]
@DeepDiver1975 
I reinstalled from scratch because I'm an idiot.

I'm on FreeBSD 10.2, and I was running php5.5, but wanted to upgrade to php5.6. During the upgrade, I somehow managed to completely mess up my system. Both ownCloud and Roundcube stopped working, and Apache 2.2 started misbehaving.

I took the opportunity to uninstall Apache 2.2, php5.5, ownCloud 8.2, and even dropped the MySQL databases for owncloud and Roundcube. I also took the opportunity to move from self signed certs to Let's Encrypt certs. I then installed Apache 2.4, php5.6, and ownCloud 9.0. Everything worked again, except I couldn't import the contacts I backed up from my previous ownCloud.

Prior to Contacts 1.2's release, I tried importing my contacts via Thunderbird, but the contacts were incomplete. Everyone was there, but no phone numbers or birthdays. I know there's been discussion regarding various fields in Contacts 1.1 and 1.2, so I'm wondering how different the fields are between clients.

My desktop is Windows 10 - is there another client I should try to import? (I'm open to just about any idea! I'm even willing to accept that I've done something to break my server.)

EDIT: And yes, the file is called "Contacts2016-02-19.vcf"

EDIT2: Outlook seems to be able to read the vcf properly, but that doesn't do me much good! ;-)

**Comment by DarkSteve** [comment]
@DeepDiver1975 
Do you have a known working test or example vcf I could try? That would demonstrate whether it is my vcf or my server that is misbehaving.

**Comment by DeepDiver1975** [comment]
Its a pure client (contacts web app) issue. Let me See what I can do ....

Please Import with any carddav client e.g. Evolution on Gnome or whatever you have.

**Comment by DarkSteve** [comment]
"Please Import with any carddav client e.g. Evolution on Gnome or whatever you have."

Bummer. My desktop/laptop is Windows, my server is headless and I interact with it via PuTTY or via web UI. Basically I have Thunderbird (which doesn't work) or my Cyanogenmod mobile (which doesn't have anything that works).

I'll keep tinkering for a while, but it looks like I'll eventually copy 'n paste almost 100 contacts from Outlook into ownCloud. I have no doubt Contacts in ownCloud will catch up to it's previous functionality and then surpass it, but for now I'm bummed out :-(

**Comment by Henni** [comment]
@aspdye blindly adding `.vcf` to the file name isn't a good idea. As there are other formats which contain contact data (e.g. `.csv`, `.xml`)

**Comment by aspdye** [comment]
Is it possible to add autodetection and add .csv, .xml import too :grin:

**Comment by Henni** [comment]
That would be the long term goal and I believe there was a proposal for GSoC but nobody took it.

**Comment by DarkSteve** [comment]
The file name extension isn't my problem, my file has the vcf extension. However, I think auto-detection of the file type is a laudable goal.

That said, it shouldn't be a high priority. We're talking about people that run their own cloud servers here, making things work as expected is the primary goal. Making things work for people that don't know what they're doing is the secondary goal. Spinning straw into gold is the tertiary goal ;-)

**Comment by rosenstein3478** [comment]
Encountered the same problem and solved it by importing my vcf files into the remote owncloud address book in thunderbird. Had to overcome a few obstacles due to missing fields and wrong charset, but finally got it working using an addon in thunderbird called MoreFunctionsForAddressBook, which was able to extract all fields from my vcf files correlty. Even encoded images have been imported successfully. Worked like a charm.

**Comment by DarkSteve** [comment]
@rosenstein3478 
I'm very interested in how you got this working. I tried using Thunderbird but gave up when virtually no phone numbers or birthdays were imported, and no images either.

I tracked down MoreFunctionsForAddressBook and installed it but I'm not sure how you used it to import the vcf properly. I'm still having the same issues. What char set and other options did you use?

EDIT: Ah! I was going to "Tools" => "Import..." and it wasn't working. However, if you create a new Addressbook and right-click => "Import vCard/vcf" then everything imports properly :-) Thanks Rosenstein!

**Comment by MeCias** [comment]
Maybe [Cardbook for Thunderbird](https://addons.mozilla.org/de/thunderbird/addon/cardbook/) can help. I haven't used it in this matter, looks promising though.

**Comment by rosenstein3478** [comment]
I created my vcf files using owncloud 8.1.6 via the web GUI of calendars app. A typical entry looks like this:

```
BEGIN:VCARD
VERSION:3.0
PRODID:-//Apple Inc.//iOS 9.2//EN
N:<last name>;<first name>;;;
FN:<first name><last name>
NICKNAME:<nick name>
EMAIL;TYPE=INTERNET,WORK,pref:<email address>
TEL;TYPE=HOME,VOICE,pref:<phone number>
TEL;TYPE=CELL,VOICE:<mobile>
ITEM1.ADR;TYPE=HOME,pref:;;<street>;<place>;;<postal code>;
BDAY:<birthday>
REV:2015-12-31T21:20:57+00:00
UID:b9d5ac3f-9957-4b1c-bf3b-a3ba9c918e75@192.168.1.3
X-MOZILLA-HTML:FALSE
END:VCARD
```

Entries themselves are separated using line breaks. Here are the steps I did to get a vcf file into owncloud:
1. Install Thunderbird-Addon morefunctionsforAB
2. Create an empty address book in owncloud
3. Create a remote address book in thunderbird linking to the one create in step 2
4. Make sure that synchronisation works, e.g. by add a dummy entry in owncloud and check whether that one appears in thunderbird after synching
5. To import the vcf file using the add on, select in address book menu "Extras -> MoreFunctionsForAddressBook -> Kontake -> vCard/vcf-Datei importieren", select the file and press OK
   ![morefunctionsforab-import-menu](https://cloud.githubusercontent.com/assets/18635051/14767093/d942b84e-0a1d-11e6-871f-f94f240715e2.png)
6. Synchronize to import contacts into owncloud

The first time you start the import, you are asked a few details like entry separator and so on. I simlpy choose the defaults. The only thing I changed afterwards was the import charset, which I set to ARMSCII-8 and converted all my vcf files to ANSI using Notepad++

![import-charset-setting](https://cloud.githubusercontent.com/assets/18635051/14767124/c4179178-0a1e-11e6-87b5-bce308195d66.png)

If your vcf files deviates, one can edit the files directly or adjust settings in morefunctionsforAB. There is the possibility to define custom fields and map them between vcf file and thunderbird.
![custom-fields](https://cloud.githubusercontent.com/assets/18635051/14767131/f745c8a8-0a1e-11e6-96a8-b34606499835.png)

**Comment by DarkSteve** [comment]
Thanks so much rosenstein3478, you've been a huge help. I've now managed to import my vCard, including images, and it's working great. All the fields came through properly, I'm very impressed. You're the best!

**Comment by TecJon** [comment]
Same problem here with Firefox 45 and Edge on Windows 10 using a file named contacts.vcf

**Comment by frederiiiic** [comment]
Same for me, i don't really have time find a way myself for the moment...

**Comment by DarkSteve** [comment]
@frederiiiic
You don't have the time? You run your own cloud server. All you have to do is read the comments in this thread, it gives you a means of getting around the issue.

**Comment by frederiiiic** [comment]
I've seen it, but spend too many times for now making things working. I'll wait a bit for a simplest way.

**Comment by ghost** [comment]
to be honest.
On the Oc8 everything was working I think for alot of users. now with the redesigned app some issues occures.
for some users I think its easier to stay on the old version instead of building workarounds with some contacts apps. Me including.
The owncloud is a great application and service and very useful for ppls who would like to store theyre own data, but who are not very familiar with finding workarounds or just dont want to note every workaround they are using.

**Comment by frederiiiic** [comment]
I've finally add my contact in my phone, and sync them with owncloud contact, that work too... pretty easy at the end.

**Comment by ghost** [comment]
What filetypes are allowed? Is VCF not supported anymore?
Because the documentation states:
Adding Contacts¶

You can add contacts using one of the following methods:

```
Import contacts using a Variant Call Format (VCF) file
Add contacts manually
```

**Comment by DarkSteve** [comment]
@sectorchan "On the Oc8 everything was working I think for alot of users. now with the redesigned app some issues occures."

Yeah, ownCloud 8.2 was excellent. It was also the version that moved to the new development model, with a new version each year and point versions in between. The software had matured to the point that it's structure had become limiting, and OC9 needed to restructure in such a way that development could continue going forward. Unfortunately that restructure has led to short term regressions.

I think of it like the move from XP to Vista. We couldn't have Win7 or 8.1 or 10 without Vista paving the way, but it was a harsh transition in the short term. OC9 has been a regression in a lot of ways, but going forward I'm glad the devs made the hard choice.

**Comment by creopard** [comment]
Same issue here with FireFox 45:
I exported my contacts from an existing OC v8.2.2 instance and wanted to import the created vcf-file to OC v9.0.1 using contacts v1.2 (filename "Kontakte.vcf").
It gives the same error as stated above "Invalid file type".

**Comment by DeepDiver1975** [comment]
Please test #392

**Comment by creopard** [comment]
#392 did it! Thanks for fixing the issue!

_(However it took me some time to understand that I had to edit the file "contacts\js\public\script.js" because the file "contacts\js\services\contact_service.js" didn't exist.)_

**Comment by DeepDiver1975** [comment]
> (However it took me some time to understand that I had to edit the file "contacts\js\public\script.js" because the file "contacts\js\services\contact_service.js" didn't exist.)

Better would have been to run make and build the app :wink:
